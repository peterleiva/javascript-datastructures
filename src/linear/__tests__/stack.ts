import Stack from "../stack";
import { shouldBehaveLikeIterable, shouldBehaveLikeStack } from "shared";

describe("Stack", () => {
	shouldBehaveLikeIterable(new Stack(1, 2, 3));
	shouldBehaveLikeStack(Stack);

	describe("Initiating", () => {
		it("push in order of arguments", () => {
			const items = [1, 2, 3, 4];
			const stack = new Stack(...items);
			expect([...stack]).toEqual(items.reverse());
		});
	});
});
