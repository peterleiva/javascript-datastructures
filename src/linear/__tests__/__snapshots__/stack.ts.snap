// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stack should behave like iterable entries return tuple of index and values 1`] = `
Array [
  Array [
    0,
    3,
  ],
  Array [
    1,
    2,
  ],
  Array [
    2,
    1,
  ],
]
`;

exports[`Stack should behave like iterable keys returns indices from collection 1`] = `
Array [
  0,
  1,
  2,
]
`;

exports[`Stack should behave like iterable must be iterator 1`] = `
Array [
  3,
  2,
  1,
]
`;

exports[`Stack should behave like iterable values returns values of collection 1`] = `
Array [
  3,
  2,
  1,
]
`;
